systemvm/patches/debian/config/opt/cloud/templates/check_heartbeat.sh.templ

```diff
@@ -22,11 +22,11 @@ STRIKE_FILE="$ROUTER_BIN_PATH/keepalived.strikes"
 
 if [ -e $ROUTER_BIN_PATH/keepalived.ts2 ]
 then
-    lasttime=$(cat $ROUTER_BIN_PATH/keepalived.ts2)
     thistime=$(cat $ROUTER_BIN_PATH/keepalived.ts)
-    diff=$(($thistime - $lasttime))
+    lasttime=$(cat $ROUTER_BIN_PATH/keepalived.ts2)
+    diff=$(($lasttime - $thistime))
     s=0
-    if [ $diff -lt 30 ]
```

Comment:
@wilderrodrigues  why there is a change in time comparison logic.
if condition will never be true with the changes last-time will be less then this-time and diff will be negative.
So this will end up filling the logs in else condition

Reply:
@jayakarteek please refer master/4.11 for latest code, why are you commenting on this very old PR?